### DISCUSSION TITLE
When were the apps code last checked?

### DISCUSSION THREAD

**Discussion by Isaacson33** [FossifyOrg/General-Discussion#99] [Q&A] [answered]
Forgive me if this is a silly question, I've little expertise in these things. I used SimpleMobileTools apps for years, they were streets ahead of the competition, and, like others, I was disappointed to find they'd been sold, then delighted to find them back in the open source realm (thanks so much!).

Given that my trust in the project was, to say the least, a little damaged by the sale - has anyone checked the code since the transfer? 

I know that the security of open source is pretty much dependant on good people such as yourselves all keeping an eye on the code to make sure it has nothing untoward in it, and non-code-savvy users such as myself have to just trust you. I had no problem with that trust until recently...

Having the main author happily sell the user-base, I'm now far less trusting that his code is safe, and I don't have the skills to check it myself.

So basically, I wanted to start that trust process again with the new team. Do you vouch for this code in these versions, it's not just blindly copied from the old versions _assuming_ it's fine, have people checked the privacy claims previously made for these apps are actually sound?

**Comment by Aga-C** [comment]
I don't know if any full audit of the code has been done, but from what I know:
- Code in SMT GitHub wasn't updated since the sale, and in the latest commits I haven't seen any suspicious code.
- F-Droid and IzzyOnDroid have their quality checks in terms of finding any anti-features. As an example, we've already removed one Google dependency from Gallery, which was discovered by IzzyOnDroid.
- Also, to be GPL-compliant, we've removed the advanced editor from Gallery.
- Apps undergo a bit of refactorings and updating before each release. That's why we still don't have all of them released, even on GitHub.
- We've forked all libraries that were already forked by Tibor, to not use his forks. In those forks, we've removed a few things related to SMT's security.

**Reply by inson1** [reply]
I wouldnt call the removed changes in the libraries security changes tbh

**Reply by Aga-C** [reply]
@inson1 It were security features from SMT point of view - it was blocking usage of these libraries in other apps. I haven't called them security change. The only security change about it, is that now we have a control over them, not Tibor.

**Reply by inson1** [reply]
@Aga-C What does security have to do with blocking use of your FORKED OPEN SOURCE library?

**Reply by inson1** [reply]
that is like greenwashing, but for open source imho

**Reply by Aga-C** [reply]
@inson1 From SMT point of view, it was security, so no one could easily fork their apps and sell them on store. 
From our point of view, moving a forked library under our hood is also a security change because we won't get any unwanted changes, we have full control over it.

**Reply by naveensingh** [reply]
@inson1 I think what @Aga-C was trying to highlight was what changes were made to the SMT code since they were forked. Those removed libraries can be considered a privacy issue because we didn't what was happening behind the doors but not having internet permission helped.

**Reply by inson1** [reply]
@naveensingh I didnt say there werent changes, I just dont like the idea calling blocking use of libraries as security
if it at least wasnt in the libraries, that were forked
I meant security for SMT, Aga-C also said it
@Aga-C Doesnt that go against the idea of GPL license and open source?

**Reply by Aga-C** [reply]
@inson1 But we removed that lock from our forks, and we don't care whether what Tibor did was good. See e.g. here: https://github.com/FossifyOrg/subsampling-scale-image-view/commit/9c43e61fdb07b2de0ab12c692e2d91a4998868a1.

**Reply by inson1** [reply]
@Aga-C I know it was removed

**Reply by dp35654** [reply]
Is it safe to remove every Simple Mobile Tools app? Looks like everything transfered perfectly with you're file's, Gallery, Phone, Contacts and Calendar which was acting up before and I don't seem to have 471 extra contacts, only thing that concerns me is it shows up in grey like a system app but it says installed by Chrome and I defiantly didn't use the play store, is this normal as I believe it's a part of Chrome code? If I'm wrong please understand I don't even know what code looks like.

**Reply by inson1** [reply]
@dp35654 Why shoudnt be it safe? But not all apps are yet released. Stock android shows app source?

**Reply by LucaMkIII** [reply]
@dp35654 It shows up gray because it wasn't installed by Google Play. It has nothing to do with whether it's a system app or not.

I use a different browser, but after I download an .apk, if I press the browser's download notification, my phone says "App installed from" my browser. If I instead install the app by opening the .apk in File Manager (which prompts it to install), it says "App installed from File Manager". Or if I get an app through F-Droid, it says "App installed from F-Droid". And something from the Play Store says "App installed from Google Play Store".

If it's from the Play Store, the text is standard, and if I press it, the app's Play Store page opens (which is a convenient way to get there). If it's installed from anywhere else, the entry is grayed out to show that it's not a pressable option.

@inson1 My phone is *almost* stock Android. If I look at my apps through Settings, in any individual app's "App Info" screen, there's a section at the bottom where it says "App installed from ______".

**Comment by naveensingh** [comment, marked as answer]
Private and Secure are two different things. 

**Private:** I have worked with pretty much all the SMT apps and I can say with some confidence that all the apps are safe to use from a privacy perspective. You don't need to worry about your personal data being stolen or worse. All Fossify apps still don't have any access to the internet as you can verify [here](https://github.com/search?q=org%3AFossifyOrg%20android.permission.INTERNET&type=code) so they can't send any data even if a piece of code wanted to. This might change if we decide to add support for RCS but even then, we won't be tracking users or selling data.

**Secure:** I would say not having the internet permission reduces the attack surface considerably and if there are any vulnerabilities or back doors in any of the apps released by Fossify, I can assure you it is not intentional.

By the way, there's a fun poll about this by Broadie Robertson: https://www.youtube.com/post/UgkxbSSrd1b1FV59p-Ry6jcaCaTAVD07Odrk

**Reply by Isaacson33** [reply]
Thanks for the explanations, and thank also to @Aga-C  for the full and open info.

What was done with these apps shook me a bit, nice to have a bit of confidence restored...
